### PR TITLE
Get CompileCommands from namedSetOfFiles

### LINKED ISSUE #11: Generation fails for more than one target
Our current configuration for github.com/magma/magma is like this with 1 binary target.

```
		"bsv.bazel.buildFlags": [
		],
		"bsv.bazel.testFlags": [
		],
		"bsv.bes.enabled": false,
		"bsv.bzl.codesearch.enabled": false,
		"bsv.bzl.invocation.buildEventPublishAllActions": false,
		"bsv.bzl.invocation.enabled": false,
		"bsv.bzl.invocation.invokeWithBuildEventStreaming": false,
		"bsv.bzl.lsp.enableCodelensStarlarkDebug": false,
		"bsv.bzl.lsp.enableCodelensRun": false,
		"bsv.bzl.remoteCache.enabled": false,
		"bsv.bzl.starlarkDebugger.enabled": false,
		// Update this field with any new targets that need compilation database generation
		"bsv.cc.compdb.targets": [
			"//lte/gateway/c/session_manager:sessiond",
		]
```

This works great as expected, but if I add another binary, I end up with an empty compile_commands.json, with not too much error to go off of.

This is the modified config that fails:
```
		"bsv.cc.compdb.targets": [
			"//lte/gateway/c/session_manager:sessiond",
                        "//lte/gateway/c/connection_tracker/src:main",
		]
```

gist of output I see: https://gist.github.com/themarwhal/b27527151f7a6732f91eb1cd3ebe9ad1

Not sure if there could be an error message somewhere else though.

Thanks!

### PR DESCRIPTION
**Pull request by wannesvanloock**
This fixes #11. For multiple target the original stderr approach does not seem to work. I noticed the same info was present in the namedSetOfFiles entries in the build event json file. I don't know why the original approach fails though.

### CODE REVIEW

File: compdb/postprocess.py [hunk 1]
```diff
@@ -47,24 +47,20 @@
...
                 print('Execution Root:', local_exec_root)
-
-    compile_command_json_db_files = []
-    for line in bazel_stderr:
-        if line.endswith('.compile_commands.json'):
-            compile_command_json_db_files.append(line.strip())
+            elif 'namedSetOfFiles' in event and 'files' in event['namedSetOfFiles']:
+                for file in event['namedSetOfFiles']['files']:
+                    if 'uri' in file and file['uri'].endswith('.compile_commands.json'):
+                        compile_command_json_db_files.add(file['uri'][7:])
```
Comment (wannesvanloock) [inline compdb/postprocess.py:63]:
```suggestion
                        compile_command_json_db_files.add(file['uri'].strip('file://'))
```
is probably cleaner.

**Comment by lucabergamini** [comment]
@pcj FYI

File: compdb/postprocess.py (hunk 1 above)
Comment (pcj) [reply compdb/postprocess.py:63]: Yes, recommend this change.  

In addition, please check that the file URI.startswith('file://').

If a user has gRPC `--remote_cache` enabled, these namedSetOfFiles URIs will have the form like `bytestream://remote-cache.example.com:443/blobs/26610f48167b10b99ad96c5b68570bcba259dc1c5e1adbd4f89e3130dacdc5e0/2655668`.  In that case it's impossible to know from the name alone this is a `compile_commands.json` file.

**Comment by pcj** [comment]
Actually I think https://github.com/stackb/bazel-stack-vscode-cc/pull/13 is a more simple fix.  For example, I always use a remote cache, and as such this would be a breaking change.

**Comment by pcj** [comment]
Parsing the namedSetOfFiles may be the most generic approach here, but I think in order to make it work with remote cache, the namedSetOfFiles need to be mapped back the correct output group of the generating target, and then also be able to download from bytestream URLs (which requires a gRPC bytestream client, which is possible to write but would significantly increase the number of dependencies needed for this extension).

**Comment by wannesvanloock** [comment]
Thanks for having a look and fixing in another way!

**Pull request closed without merging**